Elasticsearch/Search/SearchRequest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\CommonBundle\Elasticsearch\Search;
+
+use EMS\CommonBundle\Contracts\Elasticsearch\Search\SearchRequestInterface;
+
+final class SearchRequest implements SearchRequestInterface
+{
+    /** @var array */
+    private $body;
+    /** @var array */
+    private $contentTypes;
+    /** @var int */
+    private $from = 0;
+    /** @var array */
+    private $indexes;
+    /** @var int */
+    private $size = 10;
+    /** @var array */
+    private $sourceIncludes = [];
+    /** @var array */
+    private $sourceExcludes = [];
+    /** @var bool */
+    private $version = false;
+
+    public function __construct(array $indexes = [], array $contentTypes = [], array $body = [])
+    {
+        $this->body = $body;
+        $this->contentTypes = $contentTypes;
+        $this->indexes = $indexes;
+    }
+
+    public function addContentType(string $contentType): SearchRequestInterface
+    {
+        if (!in_array($contentType, $this->contentTypes, true)) {
+            $this->contentTypes[] = $contentType;
+        }
+
+        return $this;
+    }
+
+    public function addIndex(string $index): SearchRequestInterface
+    {
+        if (!in_array($index, $this->indexes, true)) {
+            $this->indexes[] = $index;
+        }
+
+        return $this;
+    }
+
+    public function addSourceInclude(string $include): SearchRequestInterface
+    {
+        if (!in_array($include, $this->sourceIncludes, true)) {
+            $this->sourceIncludes[] = $include;
+        }
+
+        return $this;
+    }
+
+    public function addSourceExclude(string $exclude): SearchRequestInterface
+    {
+        if (!in_array($exclude, $this->sourceExcludes, true)) {
+            $this->sourceExcludes[] = $exclude;
+        }
+
+        return $this;
+    }
+
+    public function getParams(): array
```

Comment:
only for client?

Reply:
Its on the interface, no need to duplicate the doc block